v2/internal/rand/seeded.go

```diff
@@ -0,0 +1,34 @@
+package rand
+
+import (
+	mathrand "math/rand"
+	"sync"
+	"time"
+)
+
+// Seeded is an interface for seeded, concurrency-safe random number generators.
+type Seeded interface {
+	// Intn returns, as an int, a non-negative pseudo-random number in [0,n). It
+	// panics if n <= 0.
+	Intn(max int) int
+}
+
+type seeded struct {
+	seededRand *mathrand.Rand
+	mut        *sync.Mutex
+}
+
+// NewSeeded returns a seeded, concurrency-safe random number generator.
+func NewSeeded() Seeded {
```

Comment:
This is a nice shim. I wish Go had this built-in.

Reply:
:sigh: I opened an issue about that... 2+ years ago.